src/app/digital/public/api/impl/DigitalComponent.ts

```diff
@@ -16,4 +19,38 @@ export class DigitalComponentImpl extends ComponentImpl<
     public override get info(): DigitalComponentInfo {
         return new DigitalComponentInfoImpl(this.circuit, this.kind);
     }
+
+    public override firstAvailable(portGroup: string): DigitalPort | undefined {
+        const Ports = this.internal.getPortsForComponent(this.id);
+
+        if (portGroup === "input") {
```

Comment:
So `portGroup` is going to be the actual `port.group`, not "input" or "output",

i.e. a Multiplexer will have the groups: "inputs", "outputs", "selects"

"inputs" and "selects" are both "Input Ports" and will both be in `inputPortGroups`
"outputs" are "Output Ports" and will be in `outputPortGroups`

the user should be able to do `myMultiplexer.firstAvailable("selects")`
which will return a different port then `myMultiplexer.firstAvailable("inputs")`